src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -4188,6 +4191,120 @@ void ReportInvalidOperation()
                 }
             }
 
+            private void PostVisitBinaryOpNodeSubtraction(BinaryOpNode node)
```

Comment:
Can much of this logic be shared with the addition path? It looks mostly the same, and not dependent on it being subtract (excpet for DateDiff cases)